torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -549,9 +549,15 @@ def prepare_for_partitioner(mod, num_primals, num_fw_outputs):
                     joint_hop_gm, num_fw_inputs, num_fw_outputs
                 )
 
+                # TODO: invoke_subgraph should track which of its inputs static indices
+                # so it can propagate them to the partitioner (and use in cudagraphs)
+                static_input_indices: list[int] = []
                 # Step 2) and 3) - Run joint graph passes and partitioner
                 new_fw_hop_gm, new_bw_hop_gm = aot_config.partition_fn(
-                    joint_hop_gm, [], num_fwd_outputs=num_fw_outputs
+                    joint_hop_gm,
+                    [],
+                    num_fwd_outputs=num_fw_outputs,
+                    static_input_indices=static_input_indices,
```

Comment:
I'm not sure static is the right word. Static can mean that the graph bakes in the value of the Tensor (like a constant). I think what we mean is "this tensor will never be freed". If that's right then I'd prefer to find some other terminology for this to avoid confusion.

Reply:
hmm that seems fair (for parameters I guess these two things are both true, but "tensor will never be freed" covers the intent better.



looking for suggestions but to put one out there: what do you think of something like "static lifetime"? (which to me indicates that the lifetime of the tensor is "static" throughout the whole program, kinda like C++ lifetimes)

Idea 1: `static_lifetime_input_indices`
Idea 2: `indices_of_inputs_that_are_never_freed` (meh)

Reply:
static_lifetime seems like a reasonable way to refer to it